packages/@aws-cdk/aws-kinesis/README.md

```diff
@@ -17,34 +17,86 @@
+[Amazon Kinesis](https://docs.aws.amazon.com/streams/latest/dev/introduction.html) provides collection and processing of large [streams](https://aws.amazon.com/streaming-data/) of data records in real time. Kinesis data streams can be used for rapid and continuous data intake and aggregation.
+
+## Table Of Contents
+- [Streams](#streams)
+  - [Encryption](#encryption)
+  - [Import](#import)
+
+
+### Streams
+
+Amazon Kinesis Data Streams ingests a large amount of data in real time, durably stores the data, and makes the data available for consumption.
+
+Using the CDK, a new Kinesis stream can be created as part of the stack using the construct's constructor. You may specify the `streamName` to give your own identifier to the stream. If not, CloudFormation will generate a name. 
+
+```ts
+new Stream(this, 'MyFirstStream', {
+    streamName: 'my-awesome-stream',
+});
+```
+
+You can also specify properties such as `shardCount` to indicate how many shards the stream should choose and a `retentionPeriod`
+to specify how many hours data in shards should remain accessible. 
 
 ```ts
-new Stream(this, 'MyFirstStream');
+new Stream(this, 'MyFirstStream', {
+    streamName: 'my-awesome-stream',
+    shardCount: 3,
+    retentionPeriod: Duration.hours(48)
+});
 ```
 
-### Encryption
+Streams are not encrypted by default.
 
-Define a KMS-encrypted stream:
+#### Encryption
+
+[Stream encryption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-kinesis-stream-streamencryption.html) enables server-side encryption using an AWS KMS key for a specified stream.
+
+You can enable encryption on your streams by specifying the `encryption` property. A KMS key will be created for you and associated with the stream. 
 
 ```ts
 const stream = new Stream(this, 'MyEncryptedStream', {
-    encryption: StreamEncryption.Kms
+    encryption: StreamEncryption.KMS
 });
-
-// you can access the encryption key:
-assert(stream.encryptionKey instanceof kms.Key);
 ```
 
-You can also supply your own key:
+You can also provide your own external KMS key to use for stream encryption by specifying the `encryptionKey` property.
 
 ```ts
-const myKmsKey = new kms.Key(this, 'MyKey');
+import * as kms from '@aws-cdk/aws-kms';
+
+const key = new kms.Key(this, 'MyKey');
 
 const stream = new Stream(this, 'MyEncryptedStream', {
-    encryption: StreamEncryption.Kms,
-    encryptionKey: myKmsKey
+    encryption: StreamEncryption.KMS,
+    encryptionKey: key
 });
+```
+
+#### Import
+
+Any Kinesis stream that has been created outside the stack can be imported into your CDK app.
+
+Streams can be imported by their ARN via the `Stream.fromStreamArn` API
+
+```ts
+const stack = new Stack (app, 'MyStack');
+
+const importedStream = Stream.fromStreamArn(stack, 'ImportedStream', 'arn:aws:kinesis:us-east-2:123456789012:stream/f3j09j2230j')
+```
+
+Encrypted Streams can also be imported by their attributes via the `Stream.fromStreamAttributes` API
```

Comment:
Do we need a separate API here? Can't we just extend the `fromStreamArn()` API with an optional kmsKey attribute?

Reply:
I think that's a good point. I'm not sure this PR is the right place to make that decision, but let's discuss it further.

It would be a breaking change but also a simplified user experience. Absolutely worth considering.